src/slam_toolbox_common.cpp

```diff
@@ -640,6 +640,7 @@ bool SlamToolbox::serializePoseGraphCallback(
   boost::mutex::scoped_lock lock(smapper_mutex_);
   serialization::write(filename, *smapper_->getMapper(),
     *dataset_, shared_from_this());
+  resp->result = resp->RESULT_SUCCESS;
```

Comment:
Is thsi really all that helpful if there's only a "success" result? If there's no code usage of any failure codes, then I don't think this service should have a success code (and if it did fail, the default `result` value would be `0` so it would appear as success.

Reply:
It results only a "success", because now this service always "successful" (give me a PoseGraph data) different from the SaveMap service. So I created Service's response structure to be corresponding.

Reply:
I don't understand. 

The point I'm making is that you goal was to add status codes to know if it failed and how. But this PR for the serialize service (not including the other save map service below) does not actually populate any error fields, therefore its not useful to have the status field at all, because it will always be successful if its received. I'm all for adding the status here if there are actually failure conditions possible to be represented and populated. 

Does that make sense? Whether you include a status or not with the PR as it is right now, you know it was successful just by its existence because there's no `else` statement to populate any other value than success (and the default value of 0 for non-populated implies success).

Reply:
Alright. I'm misunderstanding some point. I've already created a status code if there is a failure condition.(RESULT_SUCCESS = 0 and RESULT_FAILED_TO_WRITE_FILE = 255)

Reply:
My apologies. I'm misunderstanding some point. I've already created a status code if there is a failure condition.(RESULT_SUCCESS = 0 and RESULT_FAILED_TO_WRITE_FILE = 255)